tests/ui/cloned_last.rs

```diff
@@ -0,0 +1,36 @@
+// run-rustfix
+#![warn(clippy::all)]
+
+fn main() {
+    let _: Option<String> = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .last();
+
+    let _: Option<String> = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .next();
+
+    let _: usize = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .count();
+
+    let _: Vec<_> = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .take(2)
+        .collect();
+
+    let _: Vec<_> = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .skip(2)
+        .collect();
+
+    let _ = vec!["1".to_string(), "2".to_string(), "3".to_string()]
+        .iter()
+        .cloned()
+        .nth(2);
+}
```

Comment:
I'd like to see some tests against false positives if possible: Methods after `.cloned()` which are OK, and if you do anything that would require more complex tests (e.g. `.map(_)`), there should be tests for those checks, too.

Reply:
@llogiq I added some examples, where doing automatic replacement would be bad, because the signature of methods would change.

I added a list of TODOs as well.